edx_sga/static/js/src/edx_sga.js

```diff
@@ -104,7 +105,30 @@ function StaffGradedAssignmentXBlock(runtime, element) {
             updateChangeEvent(fileUpload);
         }
 
-        function renderStaffGrading(data) {
+        // function to replace the entire content of the global variable
+        // containing all the assignments
+        function replaceStudentsAssignments (data) {
+          studentAssignments = data;
+          renderStaffGrading();
+        }
+
+        // function to update one student assignment
+        function updateStudentAssignment (data) {
+          _.map(data.assignments, function (assignment) {
```

Comment:
I'm not sure what our asset pipeline looks like and if we can change this (or we get the js from edx itself) but there's a backbone compatible lodash implementation that's generally faster. 

At my last job we had both libraries available (lodash backbone-compat, and pure lodash). We used browserify instead of require, but it's probably possible in both.